card/mod/05_standard/set/abstract/attachment.rb

```diff
@@ -202,7 +207,11 @@ def mod_file?
 
 
 def assign_set_specific_attributes
-  if @set_specific && @set_specific.present?
+  # if the attachment is a empty string, just keep the content
+  attachment_name_str = attachment_name.to_s
+  if @set_specific && @set_specific.present? &&
```

Comment:
here is the evil place that makes the db_content to be `nil`. I am not sure this is a real "fix". Please comment.